tests/monitoring/monitoring_service/test_handlers.py

```diff
@@ -134,3 +144,246 @@ def test_channel_closed_event_handler_leaves_existing_channel(
 
     assert len(context.db.channels) == 1
     assert context.db.channels[0].state == ChannelState.OPENED
+
+
+def test_channel_bp_updated_event_handler_ignores_other_event(
+    context: Context,
+):
+    event = Event()
+
+    with pytest.raises(AssertionError):
+        channel_non_closing_balance_proof_updated_event_handler(
+            event=event,
+            context=context,
+        )
+
+
+def test_channel_bp_updated_event_handler_sets_update_status_if_not_set(
+    context: Context,
+):
+    event = ReceiveChannelOpenedEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        participant1='A',
+        participant2='B',
+        settle_timeout=100,
+        block_number=42,
+    )
+
+    assert len(context.db.channels) == 0
+    channel_opened_event_handler(event, context)
+
+    assert len(context.db.channels) == 1
+    assert context.db.channels[0].state == ChannelState.OPENED
+
+    event2 = ReceiveChannelClosedEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        closing_participant='B',
+        block_number=52,
+    )
+    channel_closed_event_handler(event2, context)
+
+    assert len(context.db.channels) == 1
+    assert context.db.channels[0].state == ChannelState.CLOSED
+
+    event3 = ReceiveNonClosingBalanceProofUpdatedEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        closing_participant='B',
+        nonce=2,
+        block_number=23,
+    )
+
+    channel = context.db.get_channel(event3.token_network_address, event3.channel_identifier)
+    assert channel.update_status is None
+
+    channel_non_closing_balance_proof_updated_event_handler(event3, context)
+
+    assert len(context.db.channels) == 1
+    channel = context.db.get_channel(event3.token_network_address, event3.channel_identifier)
+    assert channel.update_status is not None
+    assert channel.update_status.nonce == 2
+    assert channel.update_status.update_sender_address == 'A'
+
+    event4 = ReceiveNonClosingBalanceProofUpdatedEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        closing_participant='B',
+        nonce=5,
+        block_number=53,
+    )
+
+    channel_non_closing_balance_proof_updated_event_handler(event4, context)
+
+    assert len(context.db.channels) == 1
+    channel = context.db.get_channel(event3.token_network_address, event3.channel_identifier)
+    assert channel.update_status is not None
+    assert channel.update_status.nonce == 5
+    assert channel.update_status.update_sender_address == 'A'
+
+
+def test_monitor_new_balance_proof_event_handler_ignores_other_event(
```

Comment:
Maybe, we can have one test to test this for all event handler? This get's a bit repetitive.